### PR TITLE
Do not allow redirecting to data: or about: URLs

### PR DESCRIPTION
**Pull request by cdumez**
#### cf4ebbe5d88a1c6b84cde33660c6a693e0cdfddd
<pre>
Do not allow redirecting to data: or about: URLs
<a href="https://bugs.webkit.org/show_bug.cgi?id=230158">https://bugs.webkit.org/show_bug.cgi?id=230158</a>
&lt;rdar://83244357&gt;

Reviewed by Brent Fulgham.

Do not allow redirecting to data: or about: URLs, as per:
- <a href="https://github.com/whatwg/html/pull/7042">https://github.com/whatwg/html/pull/7042</a>

This aligns our behavior with Blink and gets us closer to Gecko.

* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/redirect-to-about.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/redirect-to-data-expected.txt:
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::willSendRequest):

Canonical link: <a href="https://commits.webkit.org/254619@main">https://commits.webkit.org/254619@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/76a3acdf1a99ab4fabc9aa8d65473897762a41e0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89576 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34127 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20283 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98868 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/155403 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/93584 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32621 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/28094 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81932 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93278 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95223 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25909 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76436 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25857 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80787 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80662 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68847 "Found 1 new API test failure: /WebKit2Gtk/TestLoaderClient:/webkit/WebKitWebPage/redirect-to-data-uri (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30378 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14741 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/30126 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15673 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3241 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33575 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38629 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32284 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34764 "Passed tests") | |

### CODE REVIEW

File: LayoutTests/http/tests/misc/redirect-to-about-blank.html [hunk 1]
```diff
@@ -1,28 +0,0 @@
-<html>
```
Comment (brentfulgham) [inline LayoutTests/http/tests/misc/redirect-to-about-blank.html]: It's nice to remove Webkit-specific tests that are covered by WPT.

**Review by brentfulgham — APPROVED** [review]
LGTM. r+

**Comment by brentfulgham** [comment]
It does look like the GTK failure is relevant:

{
            "name": "/WebKit2Gtk/TestLoaderClient:/webkit/WebKitWebPage/redirect-to-data-uri",
            "output": null
},

**Comment by cdumez** [comment]
> It does look like the GTK failure is relevant:
> 
> { "name": "/WebKit2Gtk/TestLoaderClient:/webkit/WebKitWebPage/redirect-to-data-uri", "output": null },

I dropped that test in my patch.

**Pull request closed without merging**